File: Lista 2/1/um/CanetaTesteVetor.cs
using System;

namespace um;

public class CanetaTesteVetor
{
    private List<Caneta> canetas = new List<Caneta>();

    public void cadastrarCaneta(Caneta caneta)
    {
        if (canetas.Count() < 50)
        {
            canetas.Add(caneta);
        } 
        else
        {
            Console.WriteLine("Número maximo de canetas atingido!");
        }
    }

    public void exibirCanetas()
    {
        if (canetas.Count == 0)
        {
        Console.WriteLine("Nenhuma caneta cadastrada.");
        return;
        }
    
       
        Console.WriteLine("Canetas cadastradas:");
        foreach (Caneta caneta in canetas)
        {
            Console.WriteLine(caneta.ToString()); 
        }

    }

    public void exibirQuantidade()
    {
        if (canetas.Count() == 0)
        {
            Console.WriteLine("0 canetas cadastradas");
        }
        else
        {
            Console.WriteLine($"{canetas.Count()} caneta(s) cadastradas no total!");
        }
        
    }

    public void exibirPorCor(Cores cor)
    {
        bool encontrou = false;
        foreach(var caneta in canetas)
        {
            if (caneta != null && caneta.Cor == cor)
            {
                Console.WriteLine(caneta.ToString());
                encontrou = true;
            }
        }

        if (!encontrou)
        {
            Console.WriteLine($"Nenhuma caneta da cor {cor} foi encontrada.");
        }

    }

}

File: Lista 2/1/Python/CanetaTesteVetor.py
from Caneta import Caneta

class CanetaTesteVetor:
    def __init__(self):
        self._canetas = []
        
    def cadastrar_caneta(self, caneta: Caneta):
        if len(self._canetas) < 50:
            self._canetas.append(caneta)
        else:
            print("Número maximo de canetas atingido!")
            
    def exibir_canetas(self):
        if not self._canetas:
            print("Nenhuma caneta cadastrada.")
        else:
            print("Canetas cadastradas:")
            # Forma de printar listas
            for i, caneta in enumerate(self._canetas, start=1):
                print(f"{i}. {caneta.to_string()}")

    def exibir_quantidade(self):
        if not self._canetas:
            print("0 canetas cadastradas no momento")
        else:
            print(f"{len(self._canetas)} caneta(s) cadastradas no total!")
            
    def exibir_por_cor(self,cor):
        if not self._canetas:
            print("Nenhuma caneta cadastrada para poder pesquisar!")
            return
        
        encontrou = False
        for caneta in self._canetas:
            if caneta.get_cor() == cor:  
                print(f"Caneta encontrada: {caneta.to_string()}")
                encontrou = True

        if not encontrou:
            print("Nenhuma caneta encontrada com essa cor.")
            